### DISCUSSION TITLE
🚀GitHub Copilot Metrics API GA release now available

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#141071] [Copilot News and Announcements]
### 🆕  Update: Feb 2025

Beginning February 2025, the beta Copilot `/usage` [endpoints](https://docs.github.com/en/rest/copilot/copilot-usage?apiVersion=2022-11-28) will be deprecated. [ See details](https://github.com/orgs/community/discussions/141071#discussioncomment-12299275)

### 🚀 Now Available: GitHub Copilot Metrics API in General Availability! 🚀

We’re thrilled to introduce the GA release of the **GitHub Copilot Metrics API**, now open to all GitHub Copilot Business and GitHub Copilot Enterprise users. This release is packed with features to give you greater visibility into Copilot's impact across your development workflow. Ready to dive into the data?

---

### 📊 What’s the GitHub Copilot Metrics API?

Imagine having detailed insights into how Copilot is being used across your enterprise. The Metrics API brings this to life by providing you with data on Copilot's usage—across teams, organizations, and individual features. Perfect for tracking trends, understanding usage patterns, and integrating with your own dashboards, the Metrics API helps you see where Copilot is adding value and how you can optimize its use.

### 🎉 What's New in the GA Release?

Here’s a peek at what’s included:
- **New metrics** for Pull Request summaries and Copilot Chat on GitHub.com.
- **Clarity on code completions** and Copilot Chat usage in IDEs.
- **User engagement summaries** updated daily.
- **Custom model slicing** support, coming soon!
- Data is aggregated by enterprise, organization, and team.
- Up to **28 days of historical data** available for trend analysis.

With data loaded daily (UTC) and summarized by day, you can view patterns in real-time and make more data-informed decisions.

### 🤔 Will My Current Reporting Be Affected?

To make the transition smooth, we’ve kept the Beta route online through the end of the year. This means any current reporting tied to the Beta API will continue without interruption, while you explore the newly revised schema and expanded metrics in the GA version.

### 📚 Resources to Get Started

- **Docs**: Ready to get started? [Explore the API documentation](https://docs.github.com/en/enterprise-cloud@latest/rest/copilot/copilot-metrics?apiVersion=2022-11-28) for the full breakdown of metrics and schema.
---

### 💬 Your Turn!

How are you planning to use the Copilot Metrics API? What specific metrics would give you the best insight into Copilot's impact on your projects? Whether it’s tracking usage, surfacing adoption trends, or something else entirely, we’d love to hear your ideas. Drop your thoughts below and let’s shape the future of Copilot analytics together! 👇

**Comment by AppSecCharlie** [comment]
@Akash1134 This looks great, would love to see more detailed metrics about Copilot usage and engagement. 

Question: it looks like this is disabled for my org, do I need an enterprise owner to enable? [The docs](https://docs.github.com/en/rest/copilot/copilot-metrics?apiVersion=2022-11-28#get-copilot-metrics-for-an-enterprise) are a little light but it sounds like if I get an enterprise owner to enable the Copilot Metrics API for the enterprise I can then get them for my org?

> To access this endpoint, the Copilot Metrics API access policy must be enabled or set to "no policy" for the enterprise within GitHub settings. Only owners and billing managers for the enterprise that contains the enterprise team can view Copilot metrics for the enterprise team.

**Reply by samus-aran** [reply]
Hi @charliew33,

Yes, you would need to enable at the Enterprise level. 👍

**Reply by DevOps-zhuang** [reply]
based on my test, billing manager itself is not enought, it needs owner role for the organization

**Comment by Hridoy153-cyber** [comment]
![Uploading GVbLRQZIVKWTHfCskZsPzkEGQ0t.jpg…]()

**Comment by Rose5082** [comment]
> How's it going everyone anyone interested in attending the ongoing event

- [ ]

**Comment by DevOps-zhuang** [comment]
Thanks for the GA, it looks more detailed, and indeeed more complex :-) 
like for the previous usage route, it displays the aggregate code summary data for each day, then break down it by editor/language.
```
day": "2023-10-15",
    **"total_suggestions_count": 5000,
    "total_acceptances_count": 3000,
    "total_lines_suggested": 7000,
    "total_lines_accepted": 3500,**
    "total_active_users": 15,
    "total_chat_acceptances": 45,
    "total_chat_turns": 350,
    "total_active_chat_users": 8,
```
for the GA metric route,  I was doubting whether we still need an aggregate high-leve total_code summary for each day, maybe per day per language? currently, for the copilot_ide_code_completions part,  it only has one direct KV which is "total_engaged_users": 20. in order to get the total_code_* metircs as before, we have to check the breakdown data, which is by editor/models/language, the merge them togeterh. based on my observation, the total_code_* data per day per language is a strong requirement. so it is better to add it. thanks for your consideration.

like currently, for the lanuages part under "copilot_ide_code_completions", it only display the total_engaged_users part, is it possible to add total_code_* part for each language? thanks. 

"copilot_ide_code_completions": {
      "total_engaged_users": 20,
      "languages": [
        {
          "name": "python",
          "total_engaged_users": 10
        },
        {
          "name": "ruby",
          "total_engaged_users": 10
        }
      ],

**Reply by DevOps-zhuang** [reply]
and the other consideration is about the 'engaged user' and 'active users', to be honest, I am not quite understand the difference, and I am still oberving and check the difference.

**Reply by hermic2611** [reply]
i need to knows likes to creat my our cryptomoney

**Reply by hermic2611** [reply]
I have many many difficult for great and to buy a new crytomoney

**Comment by FREDDY431** [comment]
I dint know whats happening with that guy sending u the messages he has been hacking me for like 5 years now and he has just messed my who life up i need some help to find his location please.

________________________________

**Comment by rhlbhatnagar** [comment]
Hey,
While using this API I get:

"Your organization has disabled Copilot Metrics API access. Enable it in GitHub settings to access this endpoint."

In github settings I'm unable to enable access to the metrics API as admin. I see a link to a doc on how for github pre release license terms
https://docs.github.com/en/site-policy/github-terms/github-pre-release-license-terms 

But don't actually see how to get my enterprise access to these pre release features.

**Reply by veverkap** [reply]
You need to go into your enterprise settings first (https://github.com/enterprises/{YOUR_ENTERPRISE_SLUG}/settings/copilot?tab=policies) and chose either "No Policy" or "Enabled"

![image](https://github.com/user-attachments/assets/f015d832-826f-4455-bb5d-65ffaf658c3f)
 
If you chose "Enabled", then the policy is enabled for all organizations and they cannot change the setting at the organizaiton level.

<img width="616" alt="image" src="https://github.com/user-attachments/assets/f3a4acf7-d60a-43ea-ae6a-0252cf42118a">

If you chose "No Policy", then the policy is configurable at the organization level (https://github.com/organizations/{YOUR_ORGANIZATION}/settings/copilot/policies

![image](https://github.com/user-attachments/assets/e3dc7156-d560-4366-9dc8-8d317bff1817)

**Comment by rajeshharne** [comment]
Accessing GitHub Copilot Usage API with a Business Account – Is it supported?

Question:  @Akash1134 / Team

Hello everyone,

I’m looking to track GitHub Copilot usage metrics through the API for our organization, which currently has a GitHub Business license. Specifically, I’d like to use the following endpoint:

/enterprises/{enterprise}/copilot/usage

Before diving in, I wanted to confirm: does the GitHub Business license provide access to the Copilot usage API, or is this feature exclusively available for GitHub Enterprise accounts? If it’s enterprise-only, are there any alternative methods for tracking Copilot usage in a Business account?

Thanks for any insights or guidance!

**Reply by veverkap** [reply]
Yes, a Copilot Business subscription with users is all that is needed.

**Comment by Joao-Pedro-JP** [comment]
+2

**Comment by karpikpl** [comment]
how to access new metrics?
the docs say 
> To make the transition smooth, we’ve kept the Beta route online through the end of the year. 
what's the difference between beta route and new route?

I get old data for 
x-github-api-version-selected: 2022-11-28
GET https://api.github.com/orgs/<ORG>/copilot/usage

also the link to the documentation doesn't work

**Reply by DevOps-zhuang** [reply]
copilot/usage is the old usage API. and copilot/metrics is the new one

**Reply by karpikpl** [reply]
lol, thanks. I don't know how I missed that :)

**Comment by Njem88Stars1414** [comment]
ths small my

stars1414saturn88njem14_24@hotmail.com

Sarah  red acount mobail my
awr
 MSN 11
Servar lain

**Comment by ankoalanko2** [comment]
https://github.com/user-attachments/assets/04bc9607-c0f9-4e1b-810e-388461e2b8f0

**Comment by DevOps-zhuang** [comment]
we found some data issues, the total_suggested is also 0. though there are accepated, can you double check the data quality?
the data is as below:
"copilot_ide_code_completions": {
            "editors": [
                {
                    "name": "vscode",
                    "models": [
                        {
                            "name": "default",
                            "languages": [
                                {
                                    "name": "typescript",
                                    "total_engaged_users": 1,
                                    "**total_code_acceptances": 39,
                                    "total_code_suggestions": 0,
                                    "total_code_lines_accepted": 22,
                                    "total_code_lines_suggested": 0**
                                }
                            ],
                            "is_custom_model": false,
                            "total_engaged_users": 1
                        }
                    ],
                    "total_engaged_users": 1
                }
            ],
            "languages": [
                {
                    "name": "typescript",
                    "total_engaged_users": 1
                }
            ],
            "total_engaged_users": 1
        }
    },

**Reply by DevOps-zhuang** [reply]
and "copilot_ide_code_completions": {
            "editors": [
                {
                    "name": "JetBrains",
                    "models": [
                        {
                            "name": "default",
                            "languages": [
                                {
                                    "name": "java",
                                    "total_engaged_users": 1,
                                    "total_code_acceptances": 8,
                                    "total_code_suggestions": 0,
                                    "total_code_lines_accepted": 19,
                                    "total_code_lines_suggested": 0
                                }
                            ],
                            "is_custom_model": false,
                            "total_engaged_users": 1
                        }
                    ],
                    "total_engaged_users": 1
                }
            ],

**Reply by karpikpl** [reply]
I think we've seen similar issues in sample data and API data before this was GA :(

**Reply by DevOps-zhuang** [reply]
> I think we've seen similar issues in sample data and API data before this was GA :(

Yes, we see different issues. hope it can be solved ASAP. since it is GAed, good quality is hoped.

**Comment by juusto1981** [comment]
to 21.11.2024 klo 10.25 nazifrp ***@***.***) kirjoitti:

**Comment by levanvunam** [comment]
I tried to use the  billing/seats endpoint and seems like the "assigning_team" of member is always missing, trying to export csv from the interface also show no team, anyone has similar problem?

**Reply by DevOps-zhuang** [reply]
> I tried to use the billing/seats endpoint and seems like the "assigning_team" of member is always missing, trying to export csv from the interface also show no team, anyone has similar problem?

I met the same issue before, and then found out that it is because the 'assigning_team' is the team where the user got the copilot license, that means, we need to assign the license to the individual by teams, not to the indiviual directly. ony with this way, you can get the assigning_team, it is not the team the copilot seats belong to... and I do think this design is some strange, if it can be updated to the team the user currently in, it is better. and will be align with the usage/metrics API>

**Comment by JorenVsyntra** [comment]
Hey everyone!

**Comment by Akash1134** [comment]
## Important Announcement 🔈 

We are pleased to inform you that the **GitHub Copilot Metrics Power BI app is now live and available for use**! This robust tool provides valuable insights into your GitHub Copilot usage, enabling you to monitor license assignments and track daily usage trends across your GitHub organizations. It supports various Copilot integrations, including code completion in the IDE, chat functionality within the IDE, interactions in the [GitHub.com](http://github.com/) UI, and summarizing pull requests.

### Where can I access the dashboard?
You can [download the app](https://appsource.microsoft.com/en-us/product/power-bi/github.github-copilot-metrics) from AppSource and install it in your preferred Power BI workspace. For configuration, please refer to the provided [setup instructions](https://gist.github.com/ayodejiayodele/3115d60ce7420891d2f005f7395ce99c).

### Which GitHub Copilot versions are supported?
The app supports GitHub Copilot Business and GitHub Copilot Enterprise.

Thanks!

**Comment by dharinp** [comment]
We are trying to call both /usage (old API) and /metrics (new API) - but aren't getting data for the 25th Nov and 26th Nov, across multiple Orgs. 

Is there an ongoing issue about this is being tracked internally?

**Reply by bartvd2024** [reply]
Same here, what happened?

**Reply by dharinp** [reply]
@bartvd2024 Have created a separate discussion to track this: 
https://github.com/orgs/community/discussions/145716

Hoping for some responses there!

**Reply by LilianHoeltzener** [reply]
The same for me. To my knowledge, this is the only solution we have to obtain the analytics of GHC usage. I will soon have to make a presentation in my company on the use of copilot. We need this issue resolved quickly.

**Reply by Akash1134** [reply]
Hi there, We are investigating this issue and will update once we have more details to share.

**Reply by LilianHoeltzener** [reply]
Hello there. It seems to have been a fix. Since yesterday, I'm back to the values I expected up to the day before.

**Reply by DevOps-zhuang** [reply]
for my env, data of 11/26 is still missing.

Some metrics are invalid! nonContinuousDates: [ "2024-11-26" ] invalidCodeCompletions: [ { "date": "2024-11-18", "editor": "JetBrains", "language": "java" }, { "date": "2024-11-18", "editor": "JetBrains", "language": "yaml" }, { "date": "2024-11-18", "editor": "JetBrains", "language": "xml" }, { "date": "2024-11-18", "editor": "JetBrains", "language": "ini" }, { "date": "2024-11-18", "editor": "JetBrains", "language": "go" }, { "date": "2024-11-18", "editor": "JetBrains", "language": "unknown" }, { "date": "2024-11-19", "editor": "JetBrains", "language": "java" }, { "date": "2024-11-19", "editor": "JetBrains", "language": "yaml" }, { "date": "2024-11-19", "editor": "JetBrains", "language": "go" }, { "date": "2024-11-19", "editor": "JetBrains", "language": "ini" }, { "date": "2024-11-19", "editor": "JetBrains", "language": "unknown" } ]

**Reply by ikechukwu247** [reply]
Thanks for the update

**Comment by showhand168** [comment]
Copilot is a power tool in the job?in china,how can i use it?is is free?"吱吱吱，突次"

**Comment by happylyy** [comment]
@Akash1134 
It would be very appreciate if you explain the difference **total_active_users** and **total_engaged_users** in metrics.
Also the relation of **total_active_users** and **total_engaged_users** in metrics(new) and **total_active_users**,**total_active_chat_users** in usage(old)?
The two apis return quite different data for the same day. What are the exact meanings of these Definitions?
**usage:**
"day": "2024-12-25",
        "total_suggestions_count": 27414,
        "total_acceptances_count": 8589,
        "total_lines_suggested": 78568,
        "total_lines_accepted": 13912,
        "total_active_users": 497,
        "total_chat_acceptances": 1010,
        "total_chat_turns": 5694,
        "total_active_chat_users": 435,
**matrics:**
 "date": "2024-12-25",
"total_active_users": 745,
"total_engaged_users": 680,

**Reply by happylyy** [reply]
I've figured out, "total_active_users"="copilot_ide_code_completions engaged users", "total_active_chat_users"="copilot_ide_chat engaged users".
But still confussed with "total_active_users" and "total_engaged_users"?

**Reply by Akash1134** [reply]
Hi @happylyy , The metrics provided by the Copilot Metrics API and the older usage API have some differences in terminology and possibly in their calculation methodologies. Here's an explanation of the terms and their relations:

#### Copilot Metrics API (New)
**total_active_users**: This represents the total number of users who have interacted with GitHub Copilot in any form on a given day. This can include users who received suggestions, accepted suggestions, or interacted with Copilot in other ways.
**total_engaged_users**: This represents the subset of active users who have not only interacted with Copilot but also engaged more deeply, such as by accepting suggestions or having multiple interactions.
#### Usage API (Old)
**total_active_users**: Similar to the new metrics API, this represents the total number of users who have interacted with Copilot in any form on a given day.
**total_active_chat_users:** This is a more specific metric that counts the number of users who have interacted with Copilot's chat feature specifically.
#### Comparison and Relationship:
**total_active_users (old) vs. total_active_users (new)**: Both metrics represent the total number of users interacting with Copilot. However, the new metrics API may use a different methodology or include additional types of interactions that are not captured in the old usage API, leading to differences in the reported numbers.

**total_active_chat_users (old) vs. total_engaged_users (new)**: The old metric "total_active_chat_users" specifically counts users interacting with the chat feature, whereas "total_engaged_users" in the new API represents users with deeper engagement overall, not limited to chat interactions. This broader definition can result in different counts.


Hope that helps!

**Reply by happylyy** [reply]
Thanks for your reply. It makes me get more clear with these definitions. @Akash1134 
I still have a further question about **total_active_users** and **total_engaged_users** in metrics API.
If one just chat "hello" to copilot, that meas a **total_active_users**, but not **total_engaged_users**, is that right?
While expect this kind of meaningless chat, what other conversions are used to differentiate **total_active_users** and **total_engaged_users**?

**Reply by happylyy** [reply]
Hi, @Akash1134 while I look up into the data in detail.
There are more questions~~~~
As the picture shows, for the same day, chat_turns in **metrics** and **usage** are not match!
total_engaged_users in metrics is 447, total_active_chat_users in usage is 447, this means the two terms are match.
so look into **editors**, **total_chats** in vscode is 5467,  **total_chats** in JetBrains is 3637, the sum is  9104.
but when look into the **usage**, the **total_chat_turns** is 5610! These two are match!
Could give more information about these data? Thank you very much!
![image](https://github.com/user-attachments/assets/3facf861-5718-41bc-ba34-82b9c5b5d7e5)

**Comment by Shadt12** [comment]
20892089

**Comment by Shadt12** [comment]
***@***.***

**Comment by Clas-Thurban** [comment]
Hello there!
I started using the "GitHub Copilot Metrics Power BI app" a while back but it started to act up a couple of days ago due to design or data problems in the API:
Refresh failed due to "duplicate" data received from the API.
Data source error: Column 'usage_records.date' in Table 'CopilotEnterpriseUsageNewSchema' contains a duplicate value '1/13/2025' and this is not allowed for columns on the one side of a many-to-one relationship or for columns that are used as the primary key of a table.
What can be done?

**Reply by bartvd2024** [reply]
I get the exact same issue for duplicates on the same date:

`Column '<oii>usage_records.date</oii>' in Table '<oii>CopilotEnterpriseUsageNewSchema</oii>' contains a duplicate value '<ccon>1/13/2025</ccon>' and this is not allowed for columns on the one side of a many-to-one relationship or for columns that are used as the primary key of a table.`

this integration has been running successfully for weeks before this error occurred

**Reply by matt-buchanan** [reply]
Also having the exact same error with the exact same date, had been working fine up to that point.

**Comment by DevOps-zhuang** [comment]
for the data inconsitent issue, no finding from PG at all? such issues has been since it is GAed
 {
          "name": "Xcode",
          "models": [
            {
              "name": "default",
              "languages": [
                {
                  "name": "objective-cpp",
                  "total_engaged_users": 1,
                  "total_code_acceptances": 2,
                  "total_code_suggestions": 0,
                  "total_code_lines_accepted": 0,
                  "total_code_lines_suggested": 0
                }
              ],

**Comment by mccmar** [comment]
Hi at all. I have to basically rewrite the code because the response scheme is changed from /usage to /metrics. Thanks

**Comment by audunsolemdal** [comment]
I have a question regarding privacy/anonymity

From the docs:
`This endpoint will only return results for a given day if the organization contained five or more members with active Copilot licenses on that day`

My understanding is that this is the case to preserve anonymity.

How does this work for sub-items, e.g. if a single user uses `copilot_dotcom_pull_requests` towards a specific repo. Will it be included in the metrics or is it also required that at least 5 users in that day uses copilot_dotcom_pull_requests to be listed?

```
   "copilot_dotcom_pull_requests": {
      "total_engaged_users": 1,
      "repositories": [
        {
          "name": "demo/repo1",
          "total_engaged_users": 1,
          "models": [
            {
              "name": "default",
              "is_custom_model": false,
              "custom_model_training_date": null,
              "total_pr_summaries_created": 1,
              "total_engaged_users": 1
            }
          ]
        }
    }
  }
```

**Comment by colindembovsky** [comment]
@Akash1134 I think the link to the docs in the Discussion body should be https://docs.github.com/en/enterprise-cloud@latest/rest/copilot/copilot-metrics?apiVersion=2022-11-28

**Comment by jacobpoly73** [comment]
I have enabled the Metrics setting in Github policy as below:

![image](https://github.com/user-attachments/assets/5ef6123b-7645-4839-83b8-912b1134a418)

However when I run an api call against the metrics I get as follows:

![image](https://github.com/user-attachments/assets/c039111f-2f00-4db6-963c-60b4ae90493a)

Any idea why its not recognizing the fact that I have enabled API access? Also this was enabled a few days ago.

**Reply by jacobpoly73** [reply]
ok I guess I couldnt use the org url....had to use the enterprise url instead

What didnt work: https://api.github.com/orgs/{ORG}/copilot/metrics
What worked: https://api.github.com/enterprises/{ORG}/copilot/metrics

Thanks

**Reply by jacobpoly73** [reply]
Initial feedback...The entire json structure is different from before :( All the history collected so far has to adjusted. And any development in extraction and/or powerbi reports needs to be re-done. Hmmm....

**Reply by colindembovsky** [reply]
@jacobpoly73 here's a quick [JSONata](https://jsonata.org/) snippet that you may find useful to convert `/metrics` API results back to `/usage`

```
{
    "day" : date,
    "total_suggestion_count" : $sum(copilot_ide_code_completions.editors.models["default" in name].languages.total_code_suggestions),
    "total_acceptances_count" : $sum(copilot_ide_code_completions.editors.models["default" in name].languages.total_code_acceptances),
    "total_lines_suggested" : $sum(copilot_ide_code_completions.editors.models["default" in name].languages.total_code_lines_suggested),
    "total_lines_accepted" : $sum(copilot_ide_code_completions.editors.models["default" in name].languages.total_code_lines_accepted),
    "total_active_users" : copilot_ide_code_completions.total_engaged_users,
    "total_chat_acceptances" : $sum(copilot_ide_chat.editors.models.total_chats),
    "total_chat_users" : copilot_ide_chat.total_engaged_users,
    "breakdown" : copilot_ide_code_completions.editors#$editIndex.models["default" in name].languages.{
        "language" : name,
        "editor" : $$.copilot_ide_code_completions.editors[$editIndex].name,
        "suggestion_count" : total_code_suggestions,
        "acceptance_count" : total_code_acceptances,
        "lines_suggested" : total_code_lines_suggested,
        "lines_accepted" : total_code_lines_accepted,
        "active_users" : total_engaged_users
    }
}
```

**Comment by Madeleinetaya** [comment]
Bonjour j'ai une preocupation concernant la nouvelle structure des données. j'aimerai savoir quel formule vous appliqué pour quitter de  github metrics pour obtenir  "total_active_users",
        "total_chat_acceptances"
        "total_chat_turns" de github copilot "usage"

**Comment by Akash1134** [comment]
## Summary

Beginning February 2025, the beta Copilot `/usage` [endpoints](https://docs.github.com/en/rest/copilot/copilot-usage?apiVersion=2022-11-28) will be deprecated.  

No new data will be inserted for retrieval via the beta `/usage` endpoints. This endpoint will be accessible through February, but the 28 day retention policy will remain.

### Who’s impacted?

Enterprise organizations with Copilot licenses not yet migrated from the `/usage` metrics APIs are impacted by this deprecation.

### What’s changing and why?[](#whats-changing-and-why)

The deprecation of the beta `/usage` endpoints is a part of GitHub’s effort to deliver more powerful and flexible data offerings for enterprises, organizations, and teams. The new endpoints provide:  
– Visibility into the adoption and consumption of Copilot across various stages of dev lifecycle (from code suggestions to PR reviews), from the team to the enterprise level  
– Expanded scope of metrics, with the addition of GitHub.com Copilot Chat and Copilot for Pull Requests  
– Consistent terminology with the user management API  
– Better visibility into unique users at various drilldowns

### Next steps

Ensure your organization is no longer consuming the now deprecated `/usage` endpoints in any jobs, workflows, and analytics tools.  
As an alternative to the beta Copilot `/usage` endpoints, check out the [PowerBI template](https://appsource.microsoft.com/en-us/product/power-bi/github.github-copilot-metrics?tab=Overview) and the Copilot `/metrics` [endpoints](https://docs.github.com/en/rest/copilot/copilot-metrics?apiVersion=2022-11-28).

Join the discussion in the comments below.

**Reply by Madeleinetaya** [reply]
The real issue is that I need to update the dashboard using Copilot metrics, and our entire architecture is based on the /usage structure. Instead of changing everything, I've found all the equivalences except for total_chat_turns and total_chat_acceptances, and I need to know how these are typically represented in the metrics.

**Reply by Madeleinetaya** [reply]
@Akash1134  Please can you help me beaucause i try all combinaison but none was correct

**Comment by Apokalypt** [comment]
Hello,
Recently, Github announced the fact that different models could be used in chat (GTP-4o, o1, o3-mini, ...) or even in code completion (GPT-3.5 Turbo, GPT-4o). However, I have the impression that the API doesn't detail this information ... Indeed, since we enabled it I know that some of my collaborators use “o1” while others use “GTP-4o” but in the model names returned by the /metrics API, we only see “default”.

Is this intentional or am I just not looking where I should?

**Reply by Madeleinetaya** [reply]
Still remain "default" for us until now

**Reply by Apokalypt** [reply]
That's what we see too ...
@Akash1134 is it intended ?

**Reply by aaron-seitz** [reply]
I had a question about this myself today - looking at the main post at the top, it says that "Custom model slicing support, coming soon!", so it's on their radar - hopefully they'll be rolling it out sometime soon (along with team-level data, which we're hoping for 🤞🏻)

**Reply by Apokalypt** [reply]
> I had a question about this myself today - looking at the main post at the top, it says that "Custom model slicing support, coming soon!", so it's on their radar - hopefully they'll be rolling it out sometime soon (along with team-level data, which we're hoping for 🤞🏻)

Custom model is something different already available : https://docs.github.com/en/enterprise-cloud@latest/copilot/customizing-copilot/creating-a-custom-model-for-github-copilot

I'm not talking about custom model but about models that are "offered" by Github right now 👍

**Comment by pixelxiao** [comment]
i am the owner of org, but i still can not enable it, the button is gray
![image](https://github.com/user-attachments/assets/0f9be75b-5c8c-4612-baca-3a168432ed75)
![image](https://github.com/user-attachments/assets/6ef0b4ab-194c-495b-9b9b-8463e1236885)

**Reply by karpikpl** [reply]
it has to be enabled in the enterprise for org owners to be able to select it

**Comment by radnus888** [comment]
I am using the Copilot Metrics API to fetch data, but I have noticed that it only allows fetching up to 28 days of historical data. However, I need to access the complete historical data using this API. When I tried using the Copilot Usage API to retrieve the historical data, I encountered issues. Can you help me understand whether it is possible to fetch the complete historical data, or if there are any limitations?

**Reply by Coppac** [reply]
I am having this exact same issue.  I have to keep reinstalling the App every month just to keep current data but it won't fetch data after that initial month.  How can we resolve this urgently please as this is keeping the tool from being useful?

**Comment by ingo** [comment]
Does the PowerBI dashboard allow you to see data by team?

**Reply by Coppac** [reply]
I am also interested in this.  We're looking to organize the data by team but also by users  Is there a way to do this?

**Comment by connor-dawson** [comment]
Hi,

We've setup the GitHub Copilot Metrics PowerBI app for our GitHub enterprise. In our enterprise we have two organizations. This is reflected in the app however it seems the Organization slicer on the 'Adoption Metrics', 'Usage Metrics', and 'Data Tables' pages doesn't filter the data. 

When switching between our organizations using the slicer our data is not filtered. However on the 'Org Usage Metrics' page the data is filtered when using the organization slicer.

Is this a known issue and are there plans to fix it?

Thank you

**Comment by mccmar** [comment]
hello everyone, not clear for me the difference about total_active_users (copilot/metrics), active_this_cycle/inactive_this_cycle (copilot/billing) and  last_activity_at (billing/seats). thanks in advance

**Comment by mccmar** [comment]
hi at all, I have another question. the response schema of enterprises/ENTERPRISE/copilot/metrics and ORG/copilot/metrics produce the same result. how to differentiate the organization inside the enterprise structure? thanks in advance

**Comment by jboulware** [comment]
@Akash1134 Your PowerBI app is broken since Github updated the advanced_security endpoint.  See:  https://github.com/orgs/community/discussions/145645#discussioncomment-12805328

**Comment by Coppac** [comment]
Every 28 days my Copilot Metrics fails to refresh the data and it requires it to be removed and reinstalled to continue getting the data from my environment.  I've verified the PAT used is working for other functions and hasn't changed and the error I receive is:

Data source error: DataSource.Error: Web.Contents failed to get contents from 'https://api.github.com/enterprises/vantiva-emu/settings/billing/advanced-security' (422): Unprocessable Entity. Microsoft.Data.Mashup.ErrorCode = 10117. DataSourceKind = Web. DataSourcePath = https://api.github.com/enterprises/vantiva-emu/settings/billing/advanced-security. Url = https://api.github.com/enterprises/vantiva-emu/settings/billing/advanced-security. . The exception was raised by the IDbCommand interface.
Cluster URI: WABI-US-CENTRAL-B-PRIMARY-redirect.analysis.windows.net
Activity ID: c7743cc2-74c2-3b71-852c-577f36a35ca7
Request ID: f42b6a95-d070-ea9a-07cf-132a7ad9a9b0
Time: 2025-04-21 16:52:27Z

This tool provides no value if I have to reinstall once a month.  Please provide a solution.

**Comment by gyanpenrose-kafka** [comment]
Everything was working, but then stopped. I tried updating the app, but data refresh still failed. Yesterday, I tried installing the app from scratch, but it won't connect to the data. I am not sure why or what's happening since there are no error messages.

**Reply by DevOps-zhuang** [reply]
maybe the token is outofdate? suggest to check the error message first

**Reply by nikolai-svalebaek** [reply]
I get this error again. We had this in the previous version as well. But after updating it does not work any more :(

![image](https://github.com/user-attachments/assets/198522f4-66fa-4db3-8ff2-757ac76c4a57)

**Comment by fkenji-fuji** [comment]
In Power BI Desktop it is loading the data, but when it publish to the web it is giving the following error.

`_The credentials provided for the Web source are invalid. (Source at https://api.github.com/orgs/ORG/copilot/metrics.). The exception was raised by the IDbCommand interface. Table: GH Copilot - summary._`

Please provide a solution.

**Comment by SakharamS** [comment]
Error fetching data for this visual

The expression referenced a relationship between '<oii>CopilotEnterpriseStats</oii>'[<oii>enterprise</oii>] and '<oii>EnterpriseLicenses</oii>'[<oii>enterprise</oii>], which does not hold any data because it needs to be recalculated.
Please check the technical details for more information. If you contact support, please provide these details.
Hide details
Underlying ErrorQueryUserError
Activity ID9aab78f2-1656-4e38-ac2e-80d4569bb9ca
Correlation IDb13a69f3-a0fe-de00-0abd-d960e0416b3c
Request ID18caf88d-4577-dcd0-dbed-be4af14e6d70
TimeFri Sep 05 2025 02:47:32 GMT+0530 (India Standard Time)
Service version13.0.26550.40
Client version2508.3.25682-train
Cluster URIhttps://wabi-us-east2-c-primary-redirect.analysis.windows.net/  -- I am getting this error with the latest version for this PowerBI App. I did enable the API metrics at the enterprise level.

**Comment by VictorMoni** [comment]
Amazing, thanks for the update 😁👍❤️

**Comment by itayang** [comment]
Hi, this API is great !

I have a couple of questions / suggestions:
- Now that the new dashboard is GA (https://github.blog/changelog/2025-10-28-copilot-usage-metrics-dashboard-and-api-in-public-preview/)
is there a change to get the data from the API ?
Can we have a team-level granularity there too please ?
One of the major differences in my eyes is that the dashboard includes agent usage too and the API doesn't.

Another thing that we (And I assume a lot of others) will find valuable too is metrics about actual code integration - as in:
How much code that got suggested + accepted eventually made it to production (or to a pull request / merge to main) ?
How much of it got reverted later ? 

We're tracking our teams usage and correlate the AI adoption trend with other metrics under the 'velocity' or 'quality' umbrella and the above will help us a great deal.

**Comment by mccmar** [comment]
Hi, very useful info about the user's usage.

If I understand well these new rest api integrate the previous.
Another point is that I view a lot of unknowns on feature_language and feature_model data.

Thanks
